docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+# Supporting charts on the server side
+As discussed in #367, the server side components manager and resourcifier have not yet been refactored, as of this writing, to work with [the new chart format](docs/design/chart_format.md). Presumably, expandybird should not be affected.
+
+## Context
+In discussions about the refactoring on the Slack channel, a number of potential requirements were discussed that could complicate matters, such as:
+* supporting multiple templating technologies within a single chart.
+* supporting multiple templating technologies within a single dependency graph.
+* supporting the parameterization of charts, as opposed to templates.
```

Comment:
One way to look at this is that ultimately charts have to be parameterized if templates are parameterized.  Otherwise, the parameters would be bound at the chart level, which almost completely degrades their usefulness.

I can see no reason though for multiple templates within a chart to have a different set of parameters each.  A given template should be designed for a specific chart, and sharing templates between charts seems like a bad idea.  The template should be tailored to the chart.  So even if we did have more than 1 template per chart, it makes sense to have parameters defined at the chart-level and made available to all templates via the same parameter declaration stanza (DM schema, or whatever).